### PR TITLE
Add Particle TypeScript docs

### PR DESCRIPTION
**Pull request by entorenee**
Initial docs for TypeScript in Particle. This is upcoming documentation for 10.3

### CODE REVIEW

File: particle/development/typescript-support.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# TypeScript in Particle
+
+## Getting Started
+
+As of 10.3, TypeScript is available as a development option instead of JavaScript. TypeScript brings static typing to JavaScript for more robust code and the ability to identify type errors at compile time rather than runtime. It is completely opt-in and can be used with both `.js` files as well as `.vue` files.
+
+To convert a JavaScript file to TypeScript change the file extension from `.js` to `.ts`. This will tell the Typescript compiler to parse the file and identify any type errors. Typescript file extensions are auto-resolved by Webpack and can be omitted in the import path, similar to a JavaScript module. Example: `import Component from ‘../typescript-component’;`
```
Comment (illepic) [inline particle/development/typescript-support.md]: From:

>Typescript file extensions are auto-resolved by Webpack and can be omitted in the import path

To:

>Typescript file extensions are auto-resolved by Webpack and the `.ts` extension can be omitted in the import path

File: particle/development/typescript-support.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## Add TypeScript to a Vue Single File Component (SFC)
+
+You can also add Typescript to `.vue` files to get all of the benefits of Typescript and single file components in Vue. **Note: This process differs slightly from converting a `.js` file.**
+
+In order to use Typescript in a Vue component three things must be converted:
+
+1. Add `lang="ts”` to the script tag of your component.
+2. Import Vue at the top of your script block. `import Vue from ‘vue’;`
+3. Wrap your Vue export in `Vue.extend()`
+
```
Comment (illepic) [inline particle/development/typescript-support.md:20]: Right about here would be a good point to mention/link to the clock example and show the minimal code changes.

File: particle/development/typescript-support.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- `npm run lint:ts`: This script will lint TypeScript files through TSLint and attempt to automatically resolve issues where it can. TSLint is a separate linter than ESLint and it's configuration lives at `/tslint.json`.
+- `npm run fmt:ts` This script runs Prettier with TypeScript defined as the parser and will write changes to the existing file.
+
+## Images.d.ts file
+
+Typescript relies on understanding how to parse each imported module it receives. This includes images which are handled by Webpack. To tell Typescript that these imports are modules without a specific type structure we declare them once in `tools/typings/images.d.ts`. A base level of files has already been provided but may need to be tweaked or added to on a per project basis.
+
+The Typescript compiler will let you know if it does not understand how to parse a module. If the module is a third party library look at Definitely Typed. If it is a media file, you will likely need to add it to the images definition file. The format for these declarations is `declare module '*.svg’` where ‘svg’ is the associated file extension.
```
Comment (illepic) [inline particle/development/typescript-support.md]: Link out to Definitely Typed here.

File: particle/development/typescript-support.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## How to Type Entry Files
+
+Particle already has type declarations installed for common dependencies: Node, Jest, jQuery, and Bootstrap. The root index file for a component can also be converted to TypeScript which allows for typing of custom settings that may exist. An example file can be found at `source/default/_patterns/02-molecules/carousel/index.ts`. There are a couple of changes that will be made to each file using TypeScript:
+
+- The jquery import will need to be changed from `import $ from 'jquery';` to `import * as $ from 'jquery';` for proper type resolution.
+- Any reference to `$context` will need to be annotated with the JQuery type. For example: `export function enable($context: JQuery, { carousel = {} }) {...}`. This tells TypeScript that context is of type JQuery.
```
Comment (illepic) [inline particle/development/typescript-support.md:54]: "jQuery", not "JQuery".

**Review by illepic — CHANGES_REQUESTED** [review]

File: particle/development/typescript-support.md (hunk 4 above)
Comment (entorenee) [reply particle/development/typescript-support.md:54]: TypeScript type is cased as 'JQuery'. https://github.com/phase2/particle/pull/442/files#diff-d39a55431715223b4ddef9778e08f04dR46

File: particle/development/typescript-support.md (hunk 1 above)
Comment (entorenee) [reply particle/development/typescript-support.md]: Changed.

File: particle/development/typescript-support.md (hunk 2 above)
Comment (entorenee) [reply particle/development/typescript-support.md:20]: Updated. Added a stripped down example and the path for Vue-Clock as well.

File: particle/development/typescript-support.md (hunk 3 above)
Comment (entorenee) [reply particle/development/typescript-support.md]: Added.

**Pull request closed without merging**